src/common/src/types/interval.rs

```diff
@@ -813,42 +813,45 @@ impl Display for IntervalUnit {
         let months = self.months % 12;
         let days = self.days;
         let mut space = false;
-        let mut write = |arg: std::fmt::Arguments<'_>| {
+        let mut need_pos = false;
+        let mut write_i32 = |arg: i32, unit: &str| -> std::fmt::Result {
+            if arg == 0 {
+                return Ok(());
+            }
             if space {
                 write!(f, " ")?;
             }
-            write!(f, "{arg}")?;
+            if need_pos && arg > 0 {
+                write!(f, "+")?;
+            }
+            write!(f, "{arg} {unit}")?;
+            if arg != 1 {
+                write!(f, "s")?;
+            }
             space = true;
+            need_pos = arg < 0;
```

Comment:
Why need positive sign if argument is negative?

Reply:
It is just a PostgreSQL convention. A negative part would trigger a positive sign on the next part if positive, like the test `-1 mons +1 day` we moved back from e2e slt to regress tests.

Reply:
renamed the variable to `following_neg`